src/lib/es5.d.ts

```diff
@@ -1540,6 +1540,11 @@ type Capitalize<S extends string> = intrinsic;
  */
 type Uncapitalize<S extends string> = intrinsic;
 
+/**
+ * Get a string representation of a type
+ */
+type TypeToString<T> = intrinsic;

```

Comment:
IMO, <code>[NameOf](https://github.com/microsoft/TypeScript/issues/1579)&lt;T&gt;</code> might be a better name.

Reply:
That issue requires a runtime behavior but this one is only type-level. I don't know if it is a good idea to use `NameOf`